Yoast/javascript
# Refactor html tree adapter

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Refactors the tree building process to first build an intermittent tree using the parse5 library, and afterwards transforms this into our own tree representation.

## Relevant technical choices:

* We decided to save the entire source code location, including start- and end positions of the start tag and end tag of the elements.
  * We need this information for markings and when we generate the text indices.
* Many of the unit tests for generating the tree had to be changed accordingly.
  * I decided to rewrite the unit tests to use the new `yaml` approach: describing the expected tree structure in `yaml` format, and then transforming it into an actual tree.
* #409 should be fixed by this PR.
  * I re-added the skipped comment parsing tests that failed and they pass again 🎉.
* I removed some now unused functionality:
  * The `TreeAdapter`, since we do not use the parse5 adapter approach anymore.
  * The `Ignored` element, since we now strip ignored elements from the tree.
  * The `Whitespace` element, since we never use it.

## Test instructions

This PR can be tested by following these steps:

* Start the content analysis example app.
  * Go to `javascript/apps/content-analysis` in the terminal.
  * Start the app by running `yarn start`.
* Write some HTML code in the text field in the upper-right corner of the example app.
  * Make sure that the _Build the tree_ toggle is switched to **on**.
* Check whether the built tree, inside of the _Tree_ collapsible, is what we expect it to be:
  * Text and formatting elements (`a`, `img`, `strong`, `em`, etc.) should always be encapsulated in a leaf node: a paragraph or a header.
  * The text indices of the formatting elements should be correct.
    * text indices = where in the plain text of the appropriate leaf node the formatting element start and end.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * -

## UI changes

## Quality assurance


Fixes #423
